kong-ngx-build

```diff
@@ -272,17 +272,20 @@ main() {
           "--with-http_realip_module"
           "--with-http_stub_status_module"
           "--with-http_v2_module"
-          "--with-stream_ssl_preread_module"
-          "--with-stream_realip_module"
           "-j$NPROC"
         )
-        
+
         if [ ! -z "$PCRE_VER" ]; then
           OPENRESTY_OPTS+=('--with-pcre=$PCRE_LOCATION')
         fi
 
         OPENRESTY_OPTS+=(${NGINX_EXTRA_MODULES[@]})
 
+        if [[ $OPENSSL_VER != 1.0.* ]]; then
```

Comment:
This should be dependent on OpenResty version, not OpenSSL. I think it is `!= 1.11.*`?

Reply:
You are right, my bad. So, according to this http://nginx.org/en/CHANGES they come in 1.11.4 and 1.11.5 respectively.

Should we try to do something clever with the versions or is it too much? The idea could be to only add build flags for versions that support it, and for each flag would come annotated with the version it came in.

Version parsing is always problematic though

Reply:
Having them in branches inlined here is good enough and straightforward; will you have some time to update this PR @eskerda?

Reply:
> Having them in branches inlined here is good enough and straightforward; will you have some time to update this PR @eskerda?

I meant, if we should try something that will allow us to ask "VERSION > 1.11.5". I have this `version.sh` partly cleaned up from `kong-upgrade-tests`, but still I am sure will miss something


See: [version.sh.txt](https://github.com/Kong/openresty-build-tools/files/3315550/version.sh.txt)

Reply:
For example, kong `0.13.1` will still fail, since it uses openresty `1.11.2.5`

Reply:
@eskerda It'd make sense to have such comparisons in this script, I agree. If we could avoid a new file and embed those utility functions in `kong-ngx-build` (and keep them as concise as possible), that'd be great though.